### ISSUE TITLE
Dates handled differently in v10.0.0?

### ISSUE THREAD

**Issue by loremaps** [jaydenseric/apollo-upload-client#138]
In an angular app, we use `v9.1.0` and the date variables are transformed into the ISO format e.g. `2019-01-10T22:00:00.000Z` when the request is send over the network. However, with `v10.0.0` this is not the case. The same value is send as an empty object `{}`. Is this behavior intended?

To explain it better please check this demo:
https://stackblitz.com/edit/apollo-upload-client-date-bugv10

where the following error is reported in the console:
```
"Variable "$input" got invalid value {}; Expected type String; String cannot represent a non string value: {}"
```
Now if you downgrade to `v9.1.0` it returns a successful response:
```
echoDate: "2019-01-11T16:24:27.656Z"
```
PS. If you need to peek the server it is here: https://codesandbox.io/s/nwnpyw93lm

**Comment by mike-marcacci** [comment]
Hmm, this is interesting, and is likely caused by [the change in extract-files](https://github.com/jaydenseric/extract-files/commit/09714bfe6d660451c6ab880bfd845f94f26c2489) which copies the variable tree instead of mutating it in place.

It currently treats all non-array `typeof o === "object"` types [as plain objects](https://github.com/jaydenseric/extract-files/commit/09714bfe6d660451c6ab880bfd845f94f26c2489#diff-381a9454316709fd178fcaea82c460a6R74); I imagine it should be updated to follow the [structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm), or just use [lodash's clone utility](https://lodash.com/docs/4.17.11#clone).

**Comment by slicejunk** [comment]
any workaround ?

**Comment by jaydenseric** [comment]
The date variables are meant to be of `String` type. the only reason using `Date` instances in variables on the client has worked for people so far is because the `JSON.stringify()` that happens in the HTTP link automatically runs [`.toJSON()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON#Description) on on them, which has the same effect as [`.toISOString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString).

If you make sure your date variables are strings you won't have any issues:

```
const date = new Date()
const variables = {
  date: date.toISOString()
}
```

**Comment by jaydenseric** [comment]
Just published [`extract-files@5.0.1`](https://github.com/jaydenseric/extract-files/releases/tag/v5.0.1), please let me know if after a fresh npm install everything works as expected 🙏

**Comment by loremaps** [comment]
It works great! Thank you very much! :+1: :tada:
